common/lib/xmodule/xmodule/modulestore/mixed.py

```diff
@@ -60,15 +60,27 @@ def inner(*args, **kwargs):
         def strip_key_func(val):
             """
             Strips the version and branch information according to the settings of rem_vers and rem_branch.
-            Recursively calls this function if the given value has a 'location' attribute.
+            dict contains course summary not the CourseDescriptor, check for the value type & strip
+            information accordingly.
+            Recursively calls this function if the given value has a 'location' key/attribute.
             """
             retval = val
-            if rem_vers and hasattr(retval, 'version_agnostic'):
-                retval = retval.version_agnostic()
-            if rem_branch and hasattr(retval, 'for_branch'):
-                retval = retval.for_branch(None)
-            if hasattr(retval, 'location'):
-                retval.location = strip_key_func(retval.location)
+            # If `val` is dict, it contains course summary not CourseDescriptor, handle it accordingly.
+            # Other than `dict`, `val` can be an instance of CourseDescriptor or BlockUsageLocator.
+            if isinstance(val, dict):
```

Comment:
if a `course_summary` were a `namedtuple`, you wouldn't have to change this function at all